### PR TITLE
Allow RAG to output decoder cross-attentions

### LINKED ISSUE #9468: Have RAG return generator cross-attentions when output_attentions=True
# 🚀 Have RAG return generator cross-attentions when output_attentions=True

This feature request is for the RAG code to be modified so that if `output_attentions=True`, it returns the generator's cross-attentions in addition to the attentions it already returns.

## Motivation

I'm interested in extracting the generator's attentions from a RAG generator model. Currently, `transformers` allows you to extract the generator's encoder attentions and decoder attentions, but not it's cross attentions. For example, inside `modeling_rag.py`, the return objects such as [RetrievAugLMMarginOutput](https://github.com/huggingface/transformers/blob/master/src/transformers/models/rag/modeling_rag.py#L38), have fields for these other attentions, but not the cross-attentions.

Because both T5 and BART can output cross-attentions, I think they could simply propagate up through the RAG code. Is there a reason this isn't already the case? Or could I do a PR to include the cross attentions along with the other attentions in the model output?

## Your contribution

On my own fork of `transformers`, I've already added this feature and would happily submit a PR!

### PR DESCRIPTION
**Pull request by dblakely**
# What does this PR do?

This PR makes RAG output the generator model's decoder cross-attentions when `output_attentions=True`.

Motivation and context: before this PR, RAG's output objects had attributes for the generator's encoder self-attentions and decoder self-attentions, but no option for the encoder-decoder cross-attentions. So this simply allows cross-attentions to be extracted, as well as fixing a small bug where `output_attentions` wasn't being passed into the generator.

Fixes #9468

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. Yes - #9468 
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests? - I don't believe any new tests are necessary. Existing tests pass.

## Who can review?
@patrickvonplaten, @lhoestq

### CODE REVIEW

File: src/transformers/models/rag/modeling_rag.py [hunk 1]
```diff
@@ -102,6 +102,12 @@ class RetrievAugLMMarginOutput(ModelOutput):
 
             Attentions weights of the generator decoder, after the attention softmax, used to compute the weighted
             average in the self-attention heads.
+        generator_cross_attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_attentions=True`` is passed or when ``config.output_attentions=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for each layer) of shape :obj:`(batch_size, num_heads,
+            sequence_length, sequence_length)`.
+
+            Cross-attentions weights of the generator decoder, after the attention softmax, used to compute
+            the weighted average in the cross-attention heads.
```
Comment (lhoestq) [inline src/transformers/models/rag/modeling_rag.py:110]:
```suggestion
            Cross-attentions weights of the generator decoder, after the attention softmax, used to compute the
            weighted average in the cross-attention heads.
```

File: src/transformers/models/rag/modeling_rag.py [hunk 2]
```diff
@@ -186,6 +193,12 @@ class RetrievAugLMOutput(ModelOutput):
...
```
Comment (lhoestq) [inline src/transformers/models/rag/modeling_rag.py:201]:
```suggestion
            Cross-attentions weights of the generator decoder, after the attention softmax, used to compute the
            weighted average in the cross-attention heads.
```

**Review by lhoestq — COMMENTED** [review]
Thanks ! Looks good to me :) 

The suggestions below should help you fix the CI.
I think it just comes from a line length issue

**Comment by dblakely** [comment]
@lhoestq Thanks for the suggestions! All the CI checks pass now.

**Review by lhoestq — APPROVED** [review]
Nice !
Also pinging @patrickvonplaten just to make sure

**Review by patrickvonplaten — APPROVED** [review]
Super! Thanks for you contribution

**Pull request merged**